test/css/all.js

```diff
@@ -98,21 +99,43 @@ describe(`The curated view of CSS extracts`, () => {
                 c.prop === prop && c.name === name)) {
               continue;
             }
-            it(`defines a valid ${type} "${name}"`, () => {
-              assert.doesNotThrow(() => {
-                // 2022-07-01: Drop units in numerical ranges for now since
-                // parser does not support them yet:
-                // https://github.com/csstree/csstree/issues/192
-                // (This arguably awful regexp typically replaces:
-                // <angle [-90deg,90deg]> with <angle [-90,90]>
-                // <time [0s,∞]> with <time [0,∞]>
-                const normalizedValue = desc[value].replace(
-                  /<(?<name>.*?)\s+\[\s*(?<min>\-?(?:\d+|∞))[A-Za-z]*\s*,\s*(?<max>\-?(?:\d+|∞))[A-Za-z]*\s*\]\s*>/g,
-                  '<$<name> [$<min>,$<max>]>');
-                const ast = definitionSyntax.parse(normalizedValue);
-                assert(ast.type);
-              }, `Invalid definition value: ${desc[value]}`);
-            });
+            if (type === 'descriptor') {
+              for (const dfn of desc[value]) {
+                it(`defines a valid ${type} "${dfn.name}" for at-rule "${name}"`, () => {
+                  assert.strictEqual(dfn.for, name);
+                  assert.doesNotThrow(() => {
+                    // 2022-07-01: Drop units in numerical ranges for now since
+                    // parser does not support them yet:
+                    // https://github.com/csstree/csstree/issues/192
+                    // (This arguably awful regexp typically replaces:
+                    // <angle [-90deg,90deg]> with <angle [-90,90]>
+                    // <time [0s,∞]> with <time [0,∞]>
+                    const normalizedValue = dfn.value.replace(
+                      /<(?<name>.*?)\s+\[\s*(?<min>\-?(?:\d+|∞))[A-Za-z]*\s*,\s*(?<max>\-?(?:\d+|∞))[A-Za-z]*\s*\]\s*>/g,
```

Comment:
can we DRY the Regexp? it's scary enough with a single instance…

Reply:
I didn't bother because they should disappear as soon as we move to the latest revision of css-tree (#694). Just proceeding one step at a time...

Reply:
oh, nice!